report.go

```diff
@@ -85,13 +86,22 @@ func report(files []string, typ, output string, every time.Duration) error {
 		rep, report = vegeta.NewTextReporter(&m), &m
 	case "json":
 		var m vegeta.Metrics
+		if histStr != "" {
+			m.Histogram = &vegeta.Histogram{}
+			if err := m.Histogram.Buckets.UnmarshalText([]byte(histStr)); err != nil {
+				return err
+			}
+		}
 		rep, report = vegeta.NewJSONReporter(&m), &m
 	case "hist":
-		if len(typ) < 6 {
-			return fmt.Errorf("bad buckets: '%s'", typ[4:])
-		}
 		var hist vegeta.Histogram
-		if err := hist.Buckets.UnmarshalText([]byte(typ[4:])); err != nil {
+		if histStr == "" { // Old way
+			if len(typ) < 6 {
+				return fmt.Errorf("bad buckets: '%s'", typ[4:])
+			}
+			histStr = typ[4:]
```

Comment:
Let's return an error saying that specifying the buckets via the `--type` flag is deprecated and that one should use the `--buckets` flag instead.

Reply:
Ok.